src/Microsoft.OpenApi/Validations/Rules/OpenApiPathsRules.cs

```diff
@@ -60,6 +58,38 @@ public static class OpenApiPathsRules
                     }
                 });
 
+        /// <summary>
+        ///  Replaces placeholders in the path with {}, e.g. /pets/{petId} becomes /pets/{} .
+        /// </summary>
+        /// <param name="path">The input path</param>
+        /// <returns>The path signature</returns>
+        private static string GetPathSignature(string path)
+        {
+            int i = 0;
+
+            while (i < path.Length)
```

Comment:
```suggestion
            for (var openBrace = path.IndexOf('{'); openBrace > -1;  openBrace = path.IndexOf('{', openBrace + 2))
```

Couldn't we do something like that to avoid iterating on every character but rather on each openbrace?